### PR TITLE
Add Swift

### PR DESCRIPTION
**Pull request by sergev**
Created benchmarks for the Swift programming language.

### CODE REVIEW

**Comment by AndrewBarba** [comment]
Should we match other languages with UInt32 instead? See: https://github.com/bddicken/languages/pull/88

**Comment by sergev** [comment]
> Should we match other languages with `UInt32` instead? See: #88

I don't think it makes any difference. All modern machines are 64-bit. Int32 operations are not faster than Int64.

File: fibonacci/swift/code.swift [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    }
+}
+
+func main() {
+    let u   = Int(CommandLine.arguments[1])!
+    var sum = 0
+    for i in 1..<u {
+        sum += fibonacci(n: i)
+    }
+    print(sum)
```
[the remark this answers is not in the document]
Comment (sergev) [reply fibonacci/swift/code.swift:17]: The sum is printed in all other languages. So I assume it's part of the benchmarking.

File: fibonacci/swift/code.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+func fibonacci(n: Int) -> Int {
+    if n < 2 {
+        return n
+    } else {
+        return fibonacci(n: n-1) + fibonacci(n: n-2)
+    }
+}
+
+func main() {
+    let u   = Int(CommandLine.arguments[1])!
```
[the remark this answers is not in the document]
Comment (sergev) [reply fibonacci/swift/code.swift]: Updated.

**Comment by MMP0** [comment]
Many other language benchmarks (JavaScript, PHP, R, Ruby, and Python) seem to be written inline without the `main` function. Wouldn’t it be better to do the same for Swift as a minimal example?

File: fibonacci/swift/code.swift (hunk 1 above)
[the remark this answers is not in the document]
Comment (nikolay-kapustin) [reply fibonacci/swift/code.swift:15]: doing print to console inside a test we downgrade a performance, i should to be removed to any tests

[the remark this answers is not in the document]
Comment (sergev) [reply fibonacci/swift/code.swift:15]: Enhancing other tests is not a goal of this particular PR. It should be discussed and performed separately.

**Comment by barisyild** [comment]
> Swift runs on Linux and Windows just fine

I didn't know it was possible.

File: compile.sh [hunk 3]
```diff
@@ -5,3 +5,4 @@ RUSTFLAGS="-Zlocation-detail=none" cargo +nightly build --manifest-path rust/Car
 kotlinc -include-runtime kotlin/code.kt -d kotlin/code.jar
 #kotlinc-native -include-runtime kotlin/code.kt -d kotlin/code
 dart compile exe dart/code.dart -o dart/code --target-os=macos
+swiftc -O swift/code.swift -o swift/code
```
Comment (alex-taffe) [inline compile.sh:8]: `-Ounchecked` speeds this up a fair bit vs just `-O`, but I'm not sure if it's a fair comparison to other languages. It removes force nil unwrap checking, array out of bounds checking, etc

File: loops/swift/code.swift [hunk 4]
```diff
@@ -0,0 +1,14 @@
+func main() {
+    let u = Int(CommandLine.arguments[1])!
+    let r = Int.random(in: 0...9999)            // Get a random integer 0 <= r < 10k
+    var a = [Int](repeating: 0, count: 10000)   // Array of 10k elements initialized to 0
+    for i in 0..<10000 {                        // 10k outer loop iterations
+        for j in 0..<100000 {                   // 100k inner loop iterations, per outer loop iteration
+            a[i] += j % u                       // Simple sum
+        }
+        a[i] += r                               // Add a random value to each element in array
```
[the remark this answers is not in the document]
Comment (sergev) [reply loops/swift/code.swift:9]: Makes sense! Operator ```&+=```  speeds it up by 4%.

File: fibonacci/swift/code.swift [hunk 5]
```diff
@@ -0,0 +1,18 @@
+func fibonacci(n: Int) -> Int {
+    if n < 2 {
+        return n
+    } else {
+        return fibonacci(n: n - 1) + fibonacci(n: n - 2)
```
[the remark this answers is not in the document]
Comment (sergev) [reply fibonacci/swift/code.swift:5]: I tried - it does not help much. No difference in the resulting time.

**Comment by bddicken** [comment]
Thanks! Already got swift merged from https://github.com/bddicken/languages/pull/18 so I will close this one out.

**Pull request closed without merging**